crates/ccdi-server/src/params/filter.rs

```diff
@@ -161,6 +163,18 @@ pub struct Sample {
     #[serde(default, skip_serializing_if = "Option::is_none")]
     #[param(required = false, nullable = false)]
     pub depositions: Option<String>,
+
+    /// Matches any sample where the `diagnosis` field matches the
+    /// string provided.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[param(required = false, nullable = false)]
+    pub diagnosis: Option<String>,
+
+    /// Matches any sample where the `diagnosis` field contains the
+    /// string provided, ignoring case.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[param(required = false, nullable = false)]
+    pub search: Option<String>,
```

Comment:
By adding "search" here, it is automatically available to the regular `sample` endpoint.  To restrict it to the experimental endpoint, it needs to be in a separate struct.  Clay recommended using composition + serde flatten.  Basically, a new struct after Sample called SampleDiagnosis or SampleExperimental that has a Sample, but is flattened in the serialized form.  So, perhaps something like (sorry, my rust is not good):
```suggestion
}

#[derive(Debug, Default, Deserialize, IntoParams, Introspect, Serialize)]
#[into_params(parameter_in = Query)]
pub struct SampleDiagnosis {
    #[serde(flatten)]
    pub sample: Sample,

    /// Matches any sample where the `diagnosis` field contains the
    /// string provided, ignoring case.
    #[serde(default, skip_serializing_if = "Option::is_none")]
    #[param(required = false, nullable = false)]
    pub search: Option<String>,
```

Reply:
This is really helpful. I wasn't confident in my understanding of how to use composition so I ended up copy-pasting the existing Sample query parameters into SampleDiagnosis instead. But something to look into for Subject!